File: verifier-problems/src/main/java/com/jetbrains/pluginverifier/problems/AccessType.java
package com.jetbrains.pluginverifier.problems;

import org.jetbrains.annotations.NotNull;

/**
 * Represents possible JVM visibility constants.
 *
 * @author Sergey Patrikeev
 */
public enum AccessType {
  PUBLIC("public"),
  PROTECTED("protected"),
  PACKAGE_PRIVATE("package-private"),
  PRIVATE("private");

  private final String myDescription;

  AccessType(@NotNull String description) {
    myDescription = description;
  }

  @NotNull
  public String getDescription() {
    return myDescription;
  }
}

File: verifier-core/src/main/java/com/jetbrains/pluginverifier/results/problems/AccessType.kt
package com.jetbrains.pluginverifier.results.problems

import com.jetbrains.pluginverifier.results.presentation.Presentable

enum class AccessType(private val type: String) : Presentable {
  PUBLIC("public"),
  PROTECTED("protected"),
  PACKAGE_PRIVATE("package-private"),
  PRIVATE("private");

  override fun toString(): String = type

  override val shortPresentation: String = type

  override val fullPresentation: String = type
}